hibernate-core/src/test/java/org/hibernate/orm/test/optlock/OptimisticLockTest.java

```diff
@@ -189,8 +190,9 @@ else if ( dialect instanceof CockroachDialect && ( (JDBCException) cause ).getSQ
 					"40001" ) ) {
 				// CockroachDB always runs in SERIALIZABLE isolation, and uses SQL state 40001 to indicate
 				// serialization failure.
-			}
-			else {
+			} else if ( dialect instanceof MariaDBDialect && ( (JDBCException) cause ).getErrorCode() == 1020 ) {
```

Comment:
I guess this will fail with the old exception on older MariaDB versions even with the new driver? If so, this if condition needs an additional version check.

Reply:
it will works, but has been rewritten to be clear